packages/jsreport-docx/lib/preprocess/toc.js

```diff
@@ -52,6 +54,23 @@ module.exports = (files) => {
     }
 
     if (hasTOCTitle) {
+      const originalTextNode = paragraphEl.getElementsByTagName('w:t')[0]
+
+      // pre-process headings to move `{{#if cond}}` block helper to new paragraph right before if at the very begining of heading and matching `{{/if}}` is not in the same paragraph
+      if (originalTextNode != null && originalTextNode.textContent != null && originalTextNode.textContent.startsWith("{{#if ") && !originalTextNode.textContent.includes("{{/if}}")) {
```

Comment:
We should probably compare the number of opening block helpers with closing block helper to handling situations like:

```
1. {{#if cond1}}Title{{#if cond2}} subtitle{{/if}}
Text
{{/if}}
```
```